core/src/main/java/org/jdbi/v3/Foreman.java

```diff
@@ -34,8 +37,15 @@
         this.factories.addAll(factories);
     }
 
-    Argument waffle(Class<?> expectedType, Object it, StatementContext ctx)
+    Argument waffle(Type expectedType, Object it, StatementContext ctx)
     {
+        return waffle(new TypeResolver().resolve(expectedType), it, ctx);
+    }
+
+    Argument waffle(ResolvedType expectedType, Object it, StatementContext ctx)
+    {
+        ResolvedType objectType = new TypeResolver().resolve(Object.class);
```

Comment:
It could be.

One thing I've been meaning to ask you: what would be the value of reusing the same type resolver throughout, as in your PR? Is resolving types an expensive operation?